src/JsonSchema/Constraints/BaseConstraint.php

```diff
@@ -89,4 +90,22 @@ public function reset()
     {
         $this->errors = array();
     }
+
+    /**
+     * Recursively cast an associative array to an object
+     *
+     * @param array $array
+     *
+     * @return object
+     */
+    public static function arrayToObjectRecursive($array)
+    {
```

Comment:
I would think my version is faster, because you're letting all the work happen in C, and you don't have the overhead of the recursive PHP function calls. Time it for yourself though and see what you think.